Feature/API list property integration

caioalcn opened the pull request:
# Feature
- API list property integration
- MapLocationView implementation

# Checklist
- [x]  Ao abrir Tela de Anúncios, a chamada de API para lista de anúncios deve ser feita
- [x]  Durante a chamada, a LoadingView deve ser mostrada na tela
- [x]  LoadingView deve mostrar título "Searching for listings..."
- [x]  Implementar componente em um arquivo chamado MapLocationView.swift
- [x]  O componente deve ser mostrado na Tela de Detalhes do Anúncio

Commit:
Feature:
- Integrar chamada de API na Lista de Anúncios;
- Implementar MapLocationView;

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/DataManager/APIClient/RealEstateAPIClient.swift:
```diff
@@ -11,7 +11,7 @@ protocol RealEstateAPIPropertyProtocol: AnyObject {
-public final class RealEstateAPIClient {
+public final class RealEstateAPIClient: RealEstateAPIPropertyProtocol {
     func fetchProperties(completion: @escaping (Result<[Property], ErrorRequest>) -> Void) {
         let urlString = ManagerGetURL.getPropertiesURL()
```
Injetar ManagerGetURL no init do APIClient, criar um Factory da camada da funcionalidade, que vai construir:
1. ManagetGetURL
2. RealEstateAPIClient
3. ViewModel e ViewController correspondentes
4. Injetar as dependencias atraves dos inits e o Factory retornara a ViewController com as camadas injetadas.

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyList/PropertyListFactory.swift:
```diff
@@ -0,0 +1,17 @@
+
+struct PropertyListFactory {
+    static func createPropertyList() -> PropertyListViewController {
+        let apiClient = RealEstateAPIClient()
```
Renomear APIClient de acordo com cada funcionalidade

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyList/PropertyListViewModel.swift:
```diff
@@ -0,0 +1,33 @@
+        self.apiClient = apiClient
+    }
+    
+    func fetchListProperty() {
```
Utilizar o escaping aqui, por segurança, o didFetch esta atribuido como string, sendo uma closure, e sendo chamada dentro de outra. Pode ser que tenhamos aqui um retain cycle, mesmo com uso do weak self.
Podemos usar:
1. Escaping no fetchListProperty
2. Implementar Bindable: https://www.swiftbysundell.com/articles/bindable-values-in-swift/

Se seguir a opcao 2, aprendera sobre o MVVM utilizando binds sem uso do Combine/RX e afins, de forma mais simplificada.

andrevinic commented on solutions/devsprint-andre-nogueira-1/RealEstateApp/Screens/PropertyList/PropertyTableViewCell.swift:
```diff
@@ -56,55 +57,68 @@ final class PropertyTableViewCell: UITableViewCell {
     }
-
+    
+    private func setData() {
```
implementar um metodo setup(property: Property) e neste metodo configurar com o parametro.

Commit:
Fix:
- ManagerURl inject in RealEstateAPIClientProperty;
- Renamed the apiClient reference;
- Change closure in the model to Bindble;
- Remove property value from cell and pass via func;
